### PR TITLE
Added unit tests for handler and repository

### PR DESCRIPTION
**Pull request by tarasMatskovich**
## Changelog
* Added unit tests for handler and repository.

### CODE REVIEW

File: libraryAPI/pkg/handler/books_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	t.Run("Successfully got list of books", func(t *testing.T) {
+		gin.SetMode(gin.TestMode)
+		ctrl := gomock.NewController(t)
+		m := mock.NewMockBookRepository(ctrl)
+		m.EXPECT().GetBooks().Return(mock.Books, nil)
+		handler := handler.NewHandler(service.NewService(&repository.Repository{
+			BookRepository: m,
+		}))
+		rr := httptest.NewRecorder()
+		c, _ := gin.CreateTestContext(rr)
```
Comment (diabeticMike) [inline libraryAPI/pkg/handler/books_test.go:98]: i suppose `CreateTestContext` return error, you can just check it by assert

File: libraryAPI/pkg/handler/books_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		handler.GetBooks(c)
+
+		assert.Equal(t, http.StatusOK, rr.Result().StatusCode)
+	})
+
+	t.Run("Error on get list of books", func(t *testing.T) {
+		gin.SetMode(gin.TestMode)
+		ctrl := gomock.NewController(t)
+		m := mock.NewMockBookRepository(ctrl)
+		m.EXPECT().GetBooks().Return([]model.Book{}, mock.ErrorOnGet)
```
Comment (diabeticMike) [inline libraryAPI/pkg/handler/books_test.go]:
```suggestion
		m.EXPECT().GetBooks().Return(nil, mock.ErrorOnGet)
```

File: libraryAPI/pkg/mock/books_handler.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+
+	"github.com/gin-gonic/gin"
+)
+
+func MockCreateBooksPostJson(c *gin.Context, content interface{}) {
+	c.Request.Method = "POST"
```
Comment (diabeticMike) [inline libraryAPI/pkg/mock/books_handler.go]:
```suggestion
	c.Request.Method = http.MethodPost
```

File: libraryAPI/pkg/mock/books_handler.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	c.Request.Header.Set("Content-Type", "application/json")
+
+	jsonbytes, err := json.Marshal(content)
+	if err != nil {
+		panic(err)
```
Comment (diabeticMike) [inline libraryAPI/pkg/mock/books_handler.go]: don't use panic anywhere in your code, use log.Error or some other log output

File: libraryAPI/pkg/service/book_test.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+	t.Run("Error occured on creating list of books", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		m := mock.NewMockBookRepository(ctrl)
+		m.EXPECT().CreateBooks(mock.Books).Return(0, mock.ErrorOnCreate)
+		service := service.NewBookService(m)
+
+		result, err := service.CreateBooks(mock.Books)
+
+		assert.NotNil(t, err)
```
Comment (diabeticMike) [inline libraryAPI/pkg/service/book_test.go]: try 
```suggestion
		assert.Equal(t, mock.ErrorOnCreate, err)
```

File: libraryAPI/pkg/service/book_test.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+		assert.Equal(t, 0, result)
+		assert.ErrorIs(t, err, mock.ErrorOnCreate)
```
Comment (diabeticMike) [inline libraryAPI/pkg/service/book_test.go:34]: in case you rewrite from comment above it is redundant

File: libraryAPI/pkg/service/book_test.go [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+	t.Run("Error on get list of books", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		m := mock.NewMockBookRepository(ctrl)
+		m.EXPECT().GetBooks().Return([]model.Book{}, mock.ErrorOnGet)
+		service := service.NewBookService(m)
+
+		result, err := service.GetBooks()
+
+		assert.NotNil(t, err)
```
Comment (diabeticMike) [inline libraryAPI/pkg/service/book_test.go]: same as comment above

File: libraryAPI/pkg/service/book_test.go [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+		result, err := service.GetBooks()
+
+		assert.Nil(t, err)
+		assert.Equal(t, mock.Books, result)
+	})
+
+	t.Run("Error on get list of books", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		m := mock.NewMockBookRepository(ctrl)
+		m.EXPECT().GetBooks().Return([]model.Book{}, mock.ErrorOnGet)
```
Comment (diabeticMike) [inline libraryAPI/pkg/service/book_test.go]:
```suggestion
		m.EXPECT().GetBooks().Return(nil, mock.ErrorOnGet)
```

File: libraryAPI/pkg/service/book_test.go [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+		service := service.NewBookService(m)
+
+		result, err := service.GetBooks()
+
+		assert.NotNil(t, err)
+		assert.Equal(t, []model.Book{}, result)
```
Comment (diabeticMike) [inline libraryAPI/pkg/service/book_test.go]:
```suggestion
		assert.Empty(t, result)
```

File: libraryAPI/pkg/handler/books_test.go [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+			BookRepository: m,
+		}))
+		rr := httptest.NewRecorder()
+		c, _ := gin.CreateTestContext(rr)
+		c.Request = &http.Request{
+			Header: make(http.Header),
+		}
+
+		handler.GetBooks(c)
+
```
Comment (diabeticMike) [inline libraryAPI/pkg/handler/books_test.go:126]: please also validate your response body

**Review by diabeticMike — CHANGES_REQUESTED** [review]

**Review by diabeticMike — CHANGES_REQUESTED** [review]
Minor comments + forgot to check response body

File: libraryAPI/pkg/handler/books_test.go (hunk 1 above)
Comment (tarasMatskovich) [reply libraryAPI/pkg/handler/books_test.go:98]: `CreateTestContext`  return `*gin.Engine` as second variable

**Review by diabeticMike — APPROVED** [review]

**Pull request merged**